integ-test/src/test/java/org/opensearch/sql/util/TestUtils.java

```diff
@@ -123,6 +118,18 @@ public static Response performRequest(RestClient client, Request request) {
       }
       return response;
     } catch (IOException e) {
+      if (e instanceof ResponseException
+          && ((ResponseException) e).getResponse().getStatusLine().getStatusCode() == 400
+          && e.getMessage().contains("true refresh policy is not supported.")) {
```

Comment:
I looked into specific exception we get for refresh policy related error. But today its wrapped as `ResponseException`. I think I need to have this check to make sure I'm retrying exactly for the refresh issue only. As it is a test code, if there are message updates on the exception, tests would fail and we might have to adjust the message here for that in future.